test-packages/e2e-tests/test/mail/mail.spec.ts

```diff
@@ -0,0 +1,109 @@
+import fs from 'fs';
+import { join, resolve } from 'path';
+import net from 'net';
+import nodemailer from 'nodemailer';
+import Mail from 'nodemailer/lib/mailer';
+
+describe('Mail', () => {
+  const defaultMailOptions: Mail.Options = {
+    from: '"FROM" <from@example.com>', // sender address
+    to: 'TO1@example.com, TO2@example.com', // list of receivers
+    subject: 'SUBJECT',
+    text: 'TEXT'
+  };
+
+  it('should send a mail', async () => {
+    await sendTestMail(undefined, defaultMailOptions);
+
+    const mails = fs.readdirSync(join(resolve('test'), 'mail', 'test-output'));
+    expect(
+      mails.some(mail => {
+        const mailDetails = fs.readFileSync(
+          join(resolve('test'), 'mail', 'test-output', mail),
+          { encoding: 'utf8' }
+        );
+        return (
+          mailDetails.includes('To: TO1@example.com, TO2@example.com') &&
+          mailDetails.includes('Subject: SUBJECT') &&
+          mailDetails.includes('TEXT')
+        );
+      })
+    ).toBe(true);
+  });
+
+  it('should send 100 mails', async () => {
+    const mailOptions = buildArrayWithNatualNums(100).map(
+      mailIndex =>
+        ({
+          ...defaultMailOptions,
+          subject: `mail ${mailIndex}`
+        } as Mail.Options)
+    );
+    await sendTestMail(undefined, ...mailOptions);
+    const mails = fs.readdirSync(join(resolve('test'), 'mail', 'test-output'));
+    expect(mails.length).toBeGreaterThan(99);
+  }, 60000);
+});
+
+async function sendTestMail(
+  connection?: net.Socket,
+  ...mailOptions: Mail.Options[]
+): Promise<void> {
+  // create reusable transporter object using the default SMTP transport
+  const transporter = nodemailer.createTransport({
+    connection,
+    host: 'localhost',
+    port: 5566,
+    // true for 465, false for other ports
+    secure: false,
+    auth: {
+      user: 'user',
+      pass: 'pd'
+    },
+    tls: {
+      // disable tls config to fix the self signed certificate error
+      rejectUnauthorized: false
+    }
+  });
+
+  for (const mailOptionIndex in mailOptions) {
+    // eslint-disable-next-line no-console
+    console.log(`Sending email ${mailOptionIndex}/${mailOptions.length}...`);
+    const response = await transporter.sendMail(mailOptions[mailOptionIndex]);
+    // eslint-disable-next-line no-console
+    console.log(
+      `...email ${mailOptionIndex}/${mailOptions.length} for subject "${mailOptions[mailOptionIndex].subject}" was sent successfully.`
+    );
+  }
+  transporter.close();
+  // eslint-disable-next-line no-console
+  console.log('SMTP transport connection closed.');
+}
+
+function buildArrayWithNatualNums(length): number[] {
+  return Array.from({ length }, (_, i) => i + 1);
+}
+
+// challenge
+//
+//
+// 1. sending mail
```

Comment:
Should we perhaps put this in the follow up ticket, what to do next:
- Only basic for target mail server
- onPrem with socks proxy (http also ugly and untested)
- internet with build in support (just take value from process.env) if